src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -870,6 +875,16 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                     EvidenceVerifyCallback evidenceVerifyCallback = (EvidenceVerifyCallback) callback;
 
                     evidenceVerifyCallback.setVerified(verifyEvidence(evidenceVerifyCallback.getEvidence()));
+                } else if (callback instanceof SSLCallback) {
+                    SSLCallback sslCallback = (SSLCallback) callback;
+
+                    try {
+                        X509Certificate[] x509Certificates = X500.asX509CertificateArray(sslCallback.getSslSession().getPeerCertificates());
+                        verifyEvidence(new X509PeerCertificateChainEvidence(x509Certificates));
```

Comment:
+1 we do need to follow up and tie it in with the identity cached on the session - I just want to get the general functionality in place before we add the optimisation.